advisories/2024-09-25_cometbft-bls12381-key-validation.md

```diff
@@ -0,0 +1,126 @@
+# Missing BLS12-381 Public Key Validation
+
+## Description
+
+CometBFT omitted the subgroup check on the public keys of the BLS signature scheme. The subgroup check ensures that the public key is a point on the subgroup of prime order `r` of the curve. The omission would not only have allowed for computations in arbitrary groups but, assuming a Cosmos application using the BLS12-381 keys for a key-exchange protocol, an attacker whose public key was not validated could have induced the other participant in the protocol to inadvertently perform computations with her secret key on the invalid public key, an operation which is known to leak data about the secret key [1].
```

Comment:
Move this paragraph above the Description heading. It emphasises it a bit better as an introduction.

Reply:
```suggestion
CometBFT omitted the subgroup check on the public keys of the BLS signature scheme. The subgroup check ensures that the public key is a point on the subgroup of prime order `r` of the curve. The omission would not only have allowed for computations in arbitrary groups but, assuming a Cosmos application using the BLS12-381 keys for a key-exchange protocol, an attacker whose public key was not validated could have induced the other participant in the protocol to inadvertently perform computations with their secret key on the invalid public key, an operation which is known to leak data about the secret key [1].
```